### DISCUSSION TITLE
application is not loading if i use OIDCRedirectURLsAllowed in my .conf file

### DISCUSSION THREAD

**Discussion by RamyaAyyanan** [OpenIDC/mod_auth_openidc#854] [Q&A] [answered]
Hi, i'm trying to use mod_auth_OpenIdc. sso redirection is working perfectly. but i'm facing issue while logout. 

i have installed mod_auth_openIDc using below command

RUN apt-get update && \
    apt-get install --no-install-recommends -y \
    ca-certificates libapache2-mod-auth-openidc

in my application, i have to redirect to 3rd party url when user logout. so i have used the url <Redirect_Url>?logout=<3rd party url>.  but this is returning error message "logout value "3rd party url" does not match the hostname of the current request "my application host""

i have gone through the auth_openidc.conf file and found that we need to use OIDCRedirectURLsAllowed to allow the 3rd party url.
so, i have added "OIDCRedirectURLsAllowed  <3rd party url>" in my conf file. but after that the application is not loading . no application log generated. 

could you please advise?

**Comment by zandbelt** [comment, marked as answer]
I'm guessing that you are installing an older version from the distribution repositories that does not have support for `OIDCRedirectURLsAllowed`. It would say so in the error.log.

**Reply by RamyaAyyanan** [reply]
Thank you for the swift response!!. could you please advise how can i install latest version ?

currently i'm using below command to install

RUN apt-get update &&
apt-get install --no-install-recommends -y
ca-certificates libapache2-mod-auth-openidc

**Reply by zandbelt** [reply]
you can get it from the Assets section in https://github.com/zmartzone/mod_auth_openidc/releases/tag/v2.4.11.2

**Reply by RamyaAyyanan** [reply]
Thank you!..  I've downloaded the "mod_auth_openidc-2.4.11.2.tar.gz " file and running the below command in my Docker file to install the mod_auth_opendic

COPY ./mod_auth_openidc-2.4.11.2.tar /usr/local/mod_auth_openidc-2.4.11.2.tar
RUN tar -xf /usr/local/mod_auth_openidc-2.4.11.2.tar  -C /usr/local/

is that correct?

**Reply by zandbelt** [reply]
better download the .deb package for your debian/ubuntu distro instead of compiling from source yourself

**Reply by RamyaAyyanan** [reply]
sorry i'm new to this.  i have downloaded .deb file and tried to install using docker file. 

Command to install
-------------------------
COPY ./libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb /usr/local/libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb
RUN dpkg -i /usr/local/libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb

getting below dependency error. could you please advise what i'm missing ?
----------------------------------------


 ERROR [4/5] RUN dpkg -i /usr/local/libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb                                                                                                           0.7s
------
 > [4/5] RUN dpkg -i /usr/local/libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb:
#8 0.474 Selecting previously unselected package libapache2-mod-auth-openidc.
#8 0.514 (Reading database ... 8892 files and directories currently installed.)
#8 0.515 Preparing to unpack .../libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb ...
#8 0.527 Unpacking libapache2-mod-auth-openidc (2.4.11.2-1~bullseye+1) ...
#8 0.599 dpkg: dependency problems prevent configuration of libapache2-mod-auth-openidc:
#8 0.599  libapache2-mod-auth-openidc depends on libcjose0 (>= 0.4.1); however:
#8 0.599   Package libcjose0 is not installed.
#8 0.599  libapache2-mod-auth-openidc depends on libcurl4 (>= 7.16.2); however:
#8 0.599   Package libcurl4 is not installed.
#8 0.599  libapache2-mod-auth-openidc depends on libhiredis0.14 (>= 0.14.0); however:
#8 0.599   Package libhiredis0.14 is not installed.
#8 0.599  libapache2-mod-auth-openidc depends on libjansson4 (>= 2.0.1); however:
#8 0.599   Package libjansson4 is not installed.
#8 0.599  libapache2-mod-auth-openidc depends on libpcre2-8-0 (>= 10.22); however:
#8 0.599   Package libpcre2-8-0 is not installed.
#8 0.599  libapache2-mod-auth-openidc depends on libssl1.1 (>= 1.1.0); however:
#8 0.599   Package libssl1.1 is not installed.
#8 0.599
#8 0.599 dpkg: error processing package libapache2-mod-auth-openidc (--install):
#8 0.599  dependency problems - leaving unconfigured
#8 0.632 Errors were encountered while processing:
#8 0.632  libapache2-mod-auth-openidc
------
executor failed running [/bin/sh -c dpkg -i /usr/local/libapache2-mod-auth-openidc_2.4.11.2-1.bullseye+1_amd64.deb]: exit code: 1

**Reply by RamyaAyyanan** [reply]
i have used latest ubuntu version and then tried to install provided deb file. it worked . Thanks a lot for your suppport!
